faulty_code.py

```
@@ -0,0 +1,86 @@
+        self.user = user
+        self.permissions = permissions
+        self.active_sessions.append(self)
+
+    def check_permission(self, perm):
+        """Bug: always returns True due to wrong operator."""
+        return perm or perm in self.permissions
+
+    def get_token(self):
+        """Bug: hardcoded secret, weak token generation."""
+        import hashlib
+        secret = "super_secret_key_123"
```

Ruff (0.15.13)

[error] Possible hardcoded password assigned to: "secret"

(S105)